homeassistant/components/mqtt/client.py

```diff
@@ -905,17 +909,21 @@ async def _async_perform_subscriptions(self) -> None:
         self._pending_subscriptions = {}
 
         subscription_list = list(subscriptions.items())
-        result, mid = self._mqttc.subscribe(subscription_list)
 
-        if _LOGGER.isEnabledFor(logging.DEBUG):
-            for topic, qos in subscriptions.items():
-                _LOGGER.debug("Subscribing to %s, mid: %s, qos: %s", topic, mid, qos)
-        self._last_subscribe = time.monotonic()
+        for chunk in chunked_or_all(subscription_list, MAX_SUBSCRIBES_PER_CALL):
+            result, mid = self._mqttc.subscribe(chunk)
 
-        if result == 0:
-            await self._async_wait_for_mid(mid)
-        else:
-            _raise_on_error(result)
+            if _LOGGER.isEnabledFor(logging.DEBUG):
+                for topic, qos in subscriptions.items():
+                    _LOGGER.debug(
+                        "Subscribing to %s, mid: %s, qos: %s", topic, mid, qos
+                    )
+            self._last_subscribe = time.monotonic()
+
+            if result == 0:
+                await self._async_wait_for_mid(mid)
+            else:
+                _raise_on_error(result)
```

Comment:
Realized we have this pattern in 3 places so I cleaned it up in https://github.com/home-assistant/core/pull/118067